### PR TITLE
add: FOC_DEVNET_BASEDIR support for overriding ~/.foc-devnet

### PR DESCRIPTION
**Pull request by redpanda-f**
## Configurable Base Directory with FOC_DEVNET_BASEDIR

Adds support for the `FOC_DEVNET_BASEDIR` environment variable to override the default `~/.foc-devnet` base directory. The variable supports tilde expansion for paths like `~/custom-foc-devnet`. Falls back to the default when the variable is not set. Includes the `shellexpand` crate for robust path expansion.

### CODE REVIEW

**Comment by redpanda-f** [comment]
split off from https://github.com/FilOzone/foc-devnet/pull/40 so that CI issues can be tackled separately

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ bip32 = "0.5"
 rand = "0.8"
 bls-signatures = "0.15"
 names = { version = "0.14", default-features = false }
+shellexpand = "3.0"
```
[the remark this answers is not in the document]
Comment (rvagg) [reply Cargo.toml]: needs to be reconciled, maybe:


```suggestion
shellexpand = "3"
```

File: src/paths.rs [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
-    dirs::home_dir()
-        .unwrap_or_else(|| PathBuf::from("/tmp"))
-        .join(".foc-devnet")
+    if let Ok(base_dir) = std::env::var("FOC_DEVNET_BASEDIR") {
+        PathBuf::from(shellexpand::tilde(&base_dir).as_ref())
+    } else {
+        dirs::home_dir()
+            .unwrap_or_else(|| PathBuf::from("/tmp"))
+            .join(".foc-devnet")
+    }
```
[the remark this answers is not in the document]
Comment (rvagg) [reply src/paths.rs]: wouldn't be hard to add some basic unit tests with `std::env::set_var`, I'm sure AI could spit them out - default, custom, empty string, `~` expansion

**Comment by redpanda-f** [comment]
@rvagg request re-review, added testing surrounding ENV_VAR
